--- codegen//tmp/ArtifactMapper6984683515826872725.java	2024-05-13 11:34:40.585621545 +0800
+++ user//tmp/ArtifactMapper6984683515826872725.java	2024-05-13 11:34:40.585621545 +0800
@@ -1,5 +1,7 @@
 package net.bodz.violet.schema.art.dao;
 
+import org.apache.ibatis.annotations.Param;
+
 import net.bodz.bas.db.ibatis.IEntityMapper;
 import net.bodz.violet.schema.art.Artifact;
 
@@ -7,4 +9,6 @@
         extends
             IEntityMapper<Artifact> {
 
+    Artifact selectByRfid(@Param("rfid") String rfidCode);
+
 }
