--- codegen//tmp/UserValidators12399701844262439340.ts	2025-01-16 19:45:40.553373376 +0800
+++ user//tmp/UserValidators12399701844262439340.ts	2025-01-16 19:45:40.553373376 +0800
@@ -19,12 +19,6 @@
         return this._type as UserTypeInfo;
     }
 
-    validateAttributeNames(val: Set<string>) {
-    }
-
-    validateAttributes(val: ExtraAttributes) {
-    }
-
     validateGroupIds(val: Set<int>) {
     }
 
