--- codegen//tmp/UserSecretCriteriaBuilder6209696135727456505.java	2024-05-13 11:31:58.567122159 +0800
+++ user//tmp/UserSecretCriteriaBuilder6209696135727456505.java	2024-05-13 11:31:58.567122159 +0800
@@ -3,4 +3,10 @@
 public class UserSecretCriteriaBuilder
         extends _UserSecretCriteriaBuilder_stuff<UserSecretCriteriaBuilder> {
 
+    public final StringField userName = string("\"user\".name");
+    public final StringField name = userName;
+
+    public final StringField phone = string("phone");
+    public final StringField email = string("email");
+
 }
