--- codegen//tmp/UserSecret1483097491948986122.java	2024-05-13 11:31:58.567122159 +0800
+++ user//tmp/UserSecret1483097491948986122.java	2024-05-13 11:31:58.567122159 +0800
@@ -2,10 +2,30 @@
 
 import javax.persistence.Table;
 
+import net.bodz.lily.security.IUserSecret;
+
 @Table(schema = UserSecret.SCHEMA_NAME, name = UserSecret.TABLE_NAME)
 public class UserSecret
-        extends _UserSecret_stuff {
+        extends _UserSecret_stuff
+        implements
+            IUserSecret {
 
     private static final long serialVersionUID = 1L;
 
+    public UserSecret user(User user) {
+        setUser(user);
+        return this;
+    }
+
+    public UserSecret password(String password) {
+        setPassword(password);
+        return this;
+    }
+
+    public boolean isPasswordMatched(String s) {
+        if (password == null || s == null)
+            return false;
+        return password.equals(s);
+    }
+
 }
