--- codegen//tmp/CartItem11188376278245536126.java	2024-05-13 11:34:41.545658893 +0800
+++ user//tmp/CartItem11188376278245536126.java	2024-05-13 11:34:41.545658893 +0800
@@ -1,11 +1,50 @@
 package net.bodz.violet.schema.shop;
 
+import java.math.BigDecimal;
+
 import javax.persistence.Table;
 
+import net.bodz.bas.meta.cache.Derived;
+import net.bodz.bas.meta.decl.Priority;
+import net.bodz.lily.concrete.util.IOrderItem;
+import net.bodz.violet.schema.art.Artifact;
+
+/**
+ * 购物车项
+ */
 @Table(schema = CartItem.SCHEMA_NAME, name = CartItem.TABLE_NAME)
 public class CartItem
-        extends _CartItem_stuff {
+        extends _CartItem_stuff
+        implements
+            IOrderItem {
 
     private static final long serialVersionUID = 1L;
 
+    public Artifact getArtifact() {
+        if (shopItem == null)
+            return null;
+        else
+            return shopItem.getArtifact();
+    }
+
+    @Override
+    public void setQuantity(double quantity) {
+        setQuantity(BigDecimal.valueOf(quantity));
+    }
+
+    @Override
+    public void setPrice(double price) {
+        setPrice(BigDecimal.valueOf(price));
+    }
+
+    /**
+     * 总额
+     */
+    @Override
+    @Derived
+    @Priority(202)
+    public synchronized BigDecimal getAmount() {
+        return price.multiply(quantity);
+    }
+
 }
