--- codegen//tmp/SalesOrder5679307688601887927.java	2024-05-13 11:34:43.637740237 +0800
+++ user//tmp/SalesOrder5679307688601887927.java	2024-05-13 11:34:43.637740237 +0800
@@ -1,11 +1,147 @@
 package net.bodz.violet.schema.shop;
 
+import java.math.BigDecimal;
+import java.time.OffsetDateTime;
+
 import javax.persistence.Table;
 
+import net.bodz.bas.meta.bean.DetailLevel;
+import net.bodz.bas.meta.cache.Derived;
+import net.bodz.bas.meta.decl.Redundant;
+import net.bodz.bas.repr.form.meta.OfGroup;
+import net.bodz.bas.repr.form.meta.StdGroup;
+import net.bodz.bas.repr.form.meta.StdGroup.Statistics;
+import net.bodz.lily.meta.DefaultAccessMode;
+import net.bodz.lily.security.IAccessMode;
+import net.bodz.lily.util.SizedList;
+import net.bodz.violet.schema.tran.TransportOrder;
+
+/**
+ * 订单
+ *
+ * op: 销售员/经办人
+ *
+ * owner: 制单
+ */
+@DefaultAccessMode(IAccessMode.M_COOP)
 @Table(schema = SalesOrder.SCHEMA_NAME, name = SalesOrder.TABLE_NAME)
 public class SalesOrder
         extends _SalesOrder_stuff {
 
     private static final long serialVersionUID = 1L;
 
+    private SizedList<SalesOrderItem> items = new SizedList<>();
+    private BigDecimal totalQuantity = BigDecimal.ZERO;
+    private BigDecimal totalAmount = BigDecimal.ZERO;
+
+    // make-tasks
+    // material-plans (locks)
+
+    private SizedList<TransportOrder> deliveries;
+
+    @DetailLevel(DetailLevel.HIDDEN)
+    @Derived
+    @Override
+    public OffsetDateTime getBeginTime() {
+        return super.getBeginTime();
+    }
+
+    @DetailLevel(DetailLevel.HIDDEN)
+    @Derived
+    @Override
+    public OffsetDateTime getEndTime() {
+        return super.getEndTime();
+    }
+
+    /**
+     * 下单时间
+     */
+    @OfGroup(StdGroup.Schedule.class)
+    public OffsetDateTime getOrderTime() {
+        return super.getBeginTime();
+    }
+
+    public void setOrderTime(OffsetDateTime orderTime) {
+        super.setBeginTime(orderTime);
+    }
+
+    /**
+     * 交货期限
+     */
+    @OfGroup(StdGroup.Schedule.class)
+    public OffsetDateTime getDeadline() {
+        return super.getEndTime();
+    }
+
+    public void setDeadline(OffsetDateTime deadline) {
+        super.setEndTime(deadline);
+    }
+
+    /**
+     * 明细列表
+     */
+    @DetailLevel(DetailLevel.DETAIL)
+    public SizedList<SalesOrderItem> getItems() {
+        return items;
+    }
+
+    public void setItems(SizedList<SalesOrderItem> items) {
+        this.items = items;
+    }
+
+    @Override
+    @Redundant
+    public int getLength() {
+        if (items == null)
+            return 0;
+        else
+            return items.size();
+    }
+
+    /**
+     * 总数量
+     */
+    @Override
+    @OfGroup(StdGroup.Statistics.class)
+    @Statistics
+    public BigDecimal getTotalQuantity() {
+        return totalQuantity;
+    }
+
+    @Override
+    public void setTotalQuantity(BigDecimal totalQuantity) {
+        if (totalQuantity == null)
+            throw new NullPointerException("totalQuantity");
+        this.totalQuantity = totalQuantity;
+    }
+
+    /**
+     * 总金额
+     */
+    @Override
+    @OfGroup(StdGroup.Statistics.class)
+    @Statistics
+    public BigDecimal getTotalAmount() {
+        return totalAmount;
+    }
+
+    @Override
+    public void setTotalAmount(BigDecimal totalAmount) {
+        if (totalAmount == null)
+            throw new NullPointerException("totalAmount");
+        this.totalAmount = totalAmount;
+    }
+
+    /**
+     * 送货跟踪
+     */
+    @DetailLevel(DetailLevel.DETAIL2)
+    public SizedList<TransportOrder> getDeliveries() {
+        return deliveries;
+    }
+
+    public void setDeliveries(SizedList<TransportOrder> deliveries) {
+        this.deliveries = deliveries;
+    }
+
 }
